src/os/bluestore/BlueStore.cc

```diff
@@ -3165,7 +3267,11 @@ void *BlueStore::MempoolThread::entry()
 static void aio_cb(void *priv, void *priv2)
 {
   BlueStore *store = static_cast<BlueStore*>(priv);
-  store->_txc_aio_finish(priv2);
+  if ((unsigned long long)priv2 & 0x1ull) {
+    store->deferred_aio_finish((void*)((unsigned long long)priv2 & ~1ull));
```

Comment:
What's about having more universal and less error-prone approach here -> inherit both TransContext & DeferredBatch from an abstract class, implementing single virtual callback method:
struct AioContext{
virtual void  aio_finish(BlueStore*) = 0;
};